### PR TITLE
Implementation of fuzzy logics in Matlab

### LINKED ISSUE #3563: Implementation of fuzzy logic in Matlab
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Introduction:
Fuzzy logic is a way to model logic reasoning where the truth of the statement is not binary true or false but its is a degree of truth that ranges from zero to one. In this case, zero is referred to as absolutely false while one is the absolutely true. This fuzzy logic can be used to model the fuzzy inference system which is a function that maps the input to the set of output which uses the human-interpretable rules more than extract mathematics. These types of applications are mostly used in control systems applications in which the reference and the inputs are fed in and using the set of rules based on fuzzy logic, actuating signals are produced.
Generally, fuzzy Logic are not control systems but can be used in any set of applications that requires decisions. For example, it can be used in a banking system that decides the risk of lending a loan to a person based on the personal and financial information. In this article, we will look on how to apply this in Matlab using the fuzzy logic toolbox in Matlab

### Key takeaways: 
- Understanding the fuzzy logic.
- Implementing the fuzzy logic in Matlab.
- Applications of fuzzy logic.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by collince777**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3563

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
plagiarism check done ✅

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+We need to control the speed of a motor by changing the input voltage when a set point is defined. If the motor runs fast for some reason, we need to slow it down by reducing the input voltage. Conversely, if the motor slows below the set point, the input voltage must be increased so that the motor reaches the set point.
+let the input status word be;
+Too slow
+Just right
+Too fast
+
+let the output action words be;
+Less voltage(slow down)
+No change
+More voltage(speed up)
```
Comment (srishilesh) [inline content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:48]: These statements looks a bit unclear.

Could you please rephrase them?

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+3. If the motor speed is too fast, then the voltage is less.
+
+We will define all these in a fuzzy system and create a file for this. The first step is to open the fuzzy logic toolbox. To do this, execute the `fuzzy` command in the command window. When you open the logic toolbox, a new window shown below will pop up.
+
+![fuzzy toolbox](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_three.jpeg)
+
+Here, we have three blocks, is, input, Mamdani and output. In the input, this is where we input our data and for the fuzzification process. The output section, which displays the output results, acts as a defuzzification. Finally, the Mamdani is for the rule-based block. What it means is that you define your rules in this box. Now double-click on the input to open it up. When you do so, a window shown below opens up.
+
+![input window](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_four.jpeg)
+*figure1.0*
```
Comment (srishilesh) [inline content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:62]: If possible, add these for other images as well.

(Good image description would be needed)

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+The rules base are;
+1. if the motor is running too slow, then more voltage.
+2. If the motor speed is right, then no change.
+3. If the motor speed is too fast, then the voltage is less.
+
+We will define all these in a fuzzy system and create a file for this. The first step is to open the fuzzy logic toolbox. To do this, execute the `fuzzy` command in the command window. When you open the logic toolbox, a new window shown below will pop up.
+
+![fuzzy toolbox](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_three.jpeg)
+
+Here, we have three blocks, is, input, Mamdani and output. In the input, this is where we input our data and for the fuzzification process. The output section, which displays the output results, acts as a defuzzification. Finally, the Mamdani is for the rule-based block. What it means is that you define your rules in this box. Now double-click on the input to open it up. When you do so, a window shown below opens up.
```
Comment (srishilesh) [inline content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:59]: Mamdani?

Comment (srishilesh) [reply content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:59]: What's fuzzification and defuzzification?

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+![input window](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_four.jpeg)
+*figure1.0*
+
+Here, we have three inputs `mf1`, `mf2` and `mf3`. We should now change these names according to our inputs so that they can make sense. Note that you do the renaming independently. To do it, click on the box with the current name, delete it, replace it with the name you want it to be, and click `enter`. For example, changing our `mf1` to `slow`, and the output is below.
```
Comment (srishilesh) [inline content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:64]: Always use good and understandable variable names

**Review by srishilesh — CHANGES_REQUESTED** [review]
@collince777 

I've requested for a few changes.

Please ping me when done.

Thanks.

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md (hunk 3 above)
Comment (collince777) [reply content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:68]: mamdani is a matlab block in the fuzzy logic toolbox as the statements state. Fuzzification and defuzzification are defined and explained in the overview of the fuzzy logic section

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md (hunk 4 above)
Comment (collince777) [reply content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:73]: The `mf1` and such are variable names that are already in matlab but we change them as explained in the tutorial.

**Comment by collince777** [comment]
@srishilesh changes made.

**Comment by srishilesh** [comment]
I'll start the next round of review soon.

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md (hunk 3 above)
Comment (srishilesh) [reply content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:68]: @collince777 You need not answer me here. You've to briefly explain about this in the article

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md (hunk 4 above)
Comment (srishilesh) [reply content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:73]: @collince777 Changing them would look good

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+Now, we can see that the statement's truth does not belong to a single set but can belong to multiple sets based on the degree of the truth.
+
+### Implementation of the fuzzy logic in Matlab
+Consider a very simple example;
+We need to control the speed of a motor by changing the input voltage when a set point is defined. If the motor runs fast for some reason, we need to slow it down by reducing the input voltage. Conversely, if the motor slows below the set point, the input voltage must be increased so that the motor reaches the set point.
+Below are the description for the inputs. These will help in defining the rules.
+`Too slow` 
+Just right
+Too fast
+The statements are self-explanatory.
```
Comment (srishilesh) [inline content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:51]: Yes totally.

The way of putting them together is not right.

As a beginner, the reader would find it difficult to comprehend on what you're trying to accomplish.

Have a look at few of my articles [here](https://www.section.io/engineering-education/authors/srishilesh-p-s/) on how I've explained code tutorials

File: content/articles/implementation-of-fuzzy-logic-in-matlab/index.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+![renaming](/engineering-education/implementation-of-fuzzy-logic-in-matlab/fuzzy_five.jpeg)
+*figure5.0 Rename the variables to make sense*
+
+Apply the same process for all the inputs. You just click on the curve and then rename it. Now back to our first input, `slow`. Let us now define our speed. It is done on the range box. Let's say that our speed ranges from 0 to 100, for example. So we will change the `[0 1]` vector to `[0 100]` as shown in `figure1.0`.
+We can have `n` number of rules for the `n` numbers of members of function, but for the case of simplicity, we will take these number of membership functions to be three. If we keep increasing the member's frequency of corresponding, the number of rules will keep increasing. So we let's say our slow ranges from 0 to 50. Click on that curve, and the `params` box, change the existing vector to `[0 0 50]`. The same applies to all other inputs. The second input rename to `right`, which means right speed. The range of the speed is between 50 and 100. So the vector will be `[0 50 100]`. The last input should be named `high` for high speed, and the range should be between 100 and above. The vector for this is `[50 100 100]`.
```
Comment (srishilesh) [inline content/articles/implementation-of-fuzzy-logic-in-matlab/index.md:79]: Break down this paragraph

**Review by srishilesh — CHANGES_REQUESTED** [review]
@collince777 

Looks like the suggested changes have not been resolved yet.

Please ping me after addressing them all.

**Comment by collince777** [comment]
@srishilesh changes made

**Comment by srishilesh** [comment]
@collince777 

The changes are not made yet. I would suggest you to take a very careful look at my suggestions. 

Head over to the "Files changed" tab to have a look at my suggestions.

Thanks.

**Comment by collince777** [comment]
@srishilesh changes made

**Review by srishilesh — APPROVED** [review]
@collince777 

Thanks for making the changes. Great explanation.

Make sure to go through my commits.

Moving for the final review.

Thanks

**Comment by Kayere** [comment]
Great article @collince777.
Thank you for the review @srishilesh.

**Pull request merged**